### PR TITLE
Create NSObjectExtensions

### PR DESCRIPTION
**Pull request by Tliens**
The name of the class get

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/Foundation/NSObjectExtensions [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#if canImport(Foundation) && !os(Linux)
+import Foundation
+
+// MARK: - Properties
+
+public extension NSObject{
+    /// SwifterSwift: class name.
+    class var named: String {
+        let array = NSStringFromClass(self).components(separatedBy: ".")
+        if let name =  array.last {
```
Comment (guykogus) [inline Sources/SwifterSwift/Foundation/NSObjectExtensions:10]: Return `array.last ?? ""`
You can do the same for the other function.

File: Sources/SwifterSwift/Foundation/NSObjectExtensions [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#if canImport(Foundation) && !os(Linux)
+import Foundation
+
+// MARK: - Properties
+
+public extension NSObject{
+    /// SwifterSwift: class name.
+    class var named: String {
```
Comment (guykogus) [inline Sources/SwifterSwift/Foundation/NSObjectExtensions:8]: I think `className` would be more descriptive.

**Review by guykogus — CHANGES_REQUESTED** [review]
Added a few comments. You're missing a CHANGELOG entry and unit tests.
Can you please also update the README to include the `NSObject` entry?

**Review by Tliens — COMMENTED** [review]
Change the property name

File: Sources/SwifterSwift/Foundation/NSObjectExtensions (hunk 2 above)
Comment (Tliens) [reply Sources/SwifterSwift/Foundation/NSObjectExtensions]: yeah,I modified

File: Sources/SwifterSwift/Foundation/NSObjectExtensions [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        let array = NSStringFromClass(self).components(separatedBy: ".")
+        return array.last ?? ""
+    }
+    
+    /// SwifterSwift: class name.
+    var className: String {
+        let array = NSStringFromClass(type(of: self)).components(separatedBy: ".")
+        return array.last ?? ""
+    }
+}
```
Comment (guykogus) [inline Sources/SwifterSwift/Foundation/NSObjectExtensions:18]: Aren't you missing and `#endif` at the end of the file?

**Review by guykogus — CHANGES_REQUESTED** [review]
You're still missing unit tests and a CHANGELOG entry.
You also need to add this file to the workspace (I'm guessing you've just created this by opening the package, not `SwifterSwift.xcworkspace`.
Please look at other PRs if you're not sure what that means.

**Comment by guykogus** [comment]
Closing due to inactivity. Feel free to reopen if you want to continue developing this.

**Pull request closed without merging**